source/request.test.js

```diff
@@ -0,0 +1,212 @@
+const Request = require('./request');
+
+test('preprocesses headers', () => {
+  let details = {
+    responseHeaders: [
+      {
+        name: 'server',
+        value: 'cloudflare'
+      }
+    ]
+  };
+
+  let r = new Request(details);
+
+  expect(r.headers).toMatchObject({
+    SERVER: 'cloudflare'
+  });
+});
+
+describe('Railgun', () => {
+  test('processes Railgun header v1', () => {
+    let details = {
+      responseHeaders: [
+        {
+          name: 'cf-railgun',
+          value: 'normal 96 foobar 9001'
+        }
+      ]
+    };
+
+    let r = new Request(details);
+
+    expect(r.railgunMetaData).toMatchObject({
+      normal: true,
+      id: 'normal',
+      version: '9001',
+      flags: 0x60,
+      messages: [
+        'Dictionary found in memcache',
+        'Restarted broken origin connection'
+      ]
+    });
+  });
+
+  test('processes Railgun header v2', () => {
```

Comment:
🚀 

This is great! It's something I meant to write a test for a while ago when I wrote the original Railgun header parser, there is a bunch of logic here that deserves testing.